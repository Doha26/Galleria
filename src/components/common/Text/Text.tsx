import React from 'react';
import {Text as RNText} from 'react-native';
import PropTypes from 'prop-types';

const propTypes = {
  large: PropTypes.bool,
  small: PropTypes.bool,
  numberOfLines: PropTypes.number,
  color: PropTypes.string,
  extraLarge: PropTypes.bool,
  style: PropTypes.shape({}),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

const defaultProps = {
  color: null,
  style: null,
  small: false,
  large: false,
  numberOfLines: 2,
  extraLarge: false,
  children: null,
  theme: null,
};

const Text = ({
  color,
  large,
  small,
  numberOfLines,
  theme,
  style,
  children,
  extraLarge,
  ...props
}: {
  color?: string;
  large?: boolean;
  small?: boolean;
  numberOfLines?: number;
  theme?: any;
  style: any;
  children: string | number | undefined;
  extraLarge?: boolean;
}) => {
  const textStyle = [{color: color ? color || theme.secondaryColor : null, fontSize: 18}];

  if (small) {
    textStyle.push({color: undefined, fontSize: 16});
  } else if (large) {
    textStyle.push({color: undefined, fontSize: 22});
  } else if (extraLarge) {
    textStyle.push({color: undefined, fontSize: 30});
  }

  return (
    <RNText numberOfLines={numberOfLines} {...props} style={[textStyle, style]}>
      {children}
    </RNText>
  );
};

Text.propTypes = propTypes;
Text.defaultProps = defaultProps;

export default Text;
