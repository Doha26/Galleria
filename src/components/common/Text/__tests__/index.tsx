import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Text from '~/components/common/Text/Text';

describe('Text Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Text />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have text as children', () => {
    const component = shallow(<Text>Children Text</Text>);

    expect(component.props().children).toBe('Children Text');
  });

  it('should apply style prop as an array', () => {
    const component = shallow(<Text style={[{color: 'red'}, {fontSize: 30}]}>Children Text</Text>);
    expect(component.props().style).toEqual({
      color: 'red',
      fontSize: 30,
    });
  });
});
