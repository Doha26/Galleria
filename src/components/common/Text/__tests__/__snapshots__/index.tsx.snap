// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render without issues 1`] = `
<Component
  numberOfLines={2}
  style={
    Array [
      Array [
        Object {
          "color": null,
          "fontSize": 18,
        },
      ],
      null,
    ]
  }
/>
`;
